pkg/kubefed2/federate/federate.go

```diff
@@ -308,12 +345,86 @@ func CreateFedResource(hostConfig *rest.Config, resources *fedResources, dryrun
 
 	qualifiedFedName := ctlutil.NewQualifiedName(resources.FedResource)
 	if !dryrun {
-		_, err = fedClient.Resources(resources.FedResource.GetNamespace()).Create(resources.FedResource, metav1.CreateOptions{})
+		// It might take a little while for the federated type to appear if the
+		// same is being enabled while or immediately before federating the resource.
+		err = wait.PollImmediate(createResourceRetryInterval, createResourceRetryTimeout, func() (bool, error) {
+			_, err := fedClient.Resources(resources.FedResource.GetNamespace()).Create(resources.FedResource, metav1.CreateOptions{})
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			if err != nil {
+				return false, err
+			}
+			return true, nil
+		})
+	}
+
+	glog.Infof("Successfully created %s %q from %s", fedKind, qualifiedFedName, typeConfig.GetTarget().Kind)
+	return nil
+}
+
+func createResources(cmdOut io.Writer, hostConfig *rest.Config, resourcesList []FedResources, namespace string, dryrun, enableType bool) error {
+	for _, resources := range resourcesList {
+		if enableType {
+			err := enable.CreateResources(cmdOut, hostConfig, resources.TypeResources, namespace)
```

Comment:
Have you haven't tested this with more than one resource of a given type?  `enable.CreateResources` is not idempotent when it comes to creating the FederatedTypeConfig and will fail if it already exists.

Reply:
Yes you are right, this wasn't even tested for multiple resources. The intention of this WIP was exactly to get your early feedback on my proposed impl to minimise rework.